### DISCUSSION TITLE
Building DEB packages: "mvn -P deps"

### DISCUSSION THREAD

**Discussion by scsynergy** [apache/cloudstack#9280] [Q&A] [answered]
##### ISSUE TYPE

 * Documentation Report

##### COMPONENT NAME

~~~
Building DEB packages
~~~

##### CLOUDSTACK VERSION

~~~
4.19.0.0
~~~

##### OS / ENVIRONMENT

Debian, Ubuntu ...

##### SUMMARY

The page https://docs.cloudstack.apache.org/en/4.18.1.0/installguide/building_from_source.html#building-deb-packages says to run "mvn -P deps" but the profile "deps" does not exist in the pom.xml. I assume just running "mvn" is the correct instruction?

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~
BUILD SUCCESS without any warnings or error messages
~~~

##### ACTUAL RESULTS

~~~
[WARNING] The requested profile "deps" could not be activated because it does not exist.
~~~

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by yadvr** [comment]
Hi @scsynergy thanks for sharing, I think the docs may be old/incorrect which we should fix in due course.
Meanwhile, you can refer to this on building cloudstack - https://github.com/shapeblue/hackerbook/blob/main/2-dev.md#building-cloudstack and packaging https://github.com/shapeblue/hackerbook/blob/main/2-dev.md#cloudstack-packaging

**Comment by abh1sar** [comment]
I wasn't able to reproduce the issue.
`mvn -P deps` is working for me fine. I tried with 4.19 and main on Ubunutu.

```
00:01 cloudstack $ mvn -P deps                                                                                                               
[INFO] Scanning for projects...                                                                                                              
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:                                                                                                                  
[INFO]                                                                                                                                       
[INFO] Apache CloudStack Developer Tools - Checkstyle Configuration       [jar]
[INFO] Apache CloudStack                                                  [pom]     
...
```

@scsynergy can you please check if this is still an issue.
cc @rohityadavcloud

**Reply by scsynergy** [reply]
The issue occurred while building Cloudstack 4.19.0.0 on a Debian 12 bullseye machine, which is why I always used `'dpkg-buildpackage -uc -us -b'` to build the project.

And the file /opt/cloudstack-sources/debian/control has some dependency issues:
"| mysql-connector-python-py3" is the only connector available as of now, but it is missing as an alternative
a "|" character is missing between "python-setuptools" and "python3-setuptools"

**Reply by yadvr** [reply, marked as answer]
Debian 12 isn’t a support build env. You may use Ubuntu 20.04/22.04 VM or container image.

**Reply by weizhouapache** [reply]
> The issue occurred while building Cloudstack 4.19.0.0 on a Debian 12 bullseye machine, which is why I always used `'dpkg-buildpackage -uc -us -b'` to build the project.
> 
> And the file /opt/cloudstack-sources/debian/control has some dependency issues: "| mysql-connector-python-py3" is the only connector available as of now, but it is missing as an alternative a "|" character is missing between "python-setuptools" and "python3-setuptools"

@scsynergy 
as described in #9738, you can install the python3-mysql.connector package for ubuntu
```
   wget http://launchpadlibrarian.net/717081070/python3-mysql.connector_8.0.15-4_all.deb
   DEBIAN_FRONTEND=noninteractive apt install -y /tmp/python3-mysql.connector_8.0.15-4_all.deb
```
